### PR TITLE
[feg][cwf][orc8r]Promote script to promote docker images from test repo to prod repo

### PR DESCRIPTION
**Pull request by ranj8work**
Signed-off-by: Ranjini Iyer <ranj8work@gmail.com>

## Summary
**First version of Promote script to promote docker images of feg, cwf, orc8r or docker from -test repo to -prod repo**
Added changes to push helm charts to the new repo as well.

## Test Plan
Testing is done by running this script on httpd docker image. Promoted its version 1.0 from feg-test to feg-prod. Screenshots are attached below: 

image
image

Testing for changes to push helm charts to the new repo:
image

## Additional Information

Changes made after the review comments from @tmdzk 
Screenshots of test evidence are captured below:
image

image

image

### CODE REVIEW

**Comment by ranj8work** [comment]
Updated changes to push helm charts to new jfrog as well:

Testing screenshot is pasted below:

image

File: orc8r/tools/helm/package.sh
Comment (tmdzk) [inline orc8r/tools/helm/package.sh]: Let's separate the Helm changes from the more general promotion changes.

File: orc8r/tools/helm/package.sh
Comment (tmdzk) [inline orc8r/tools/helm/package.sh]: This script is meant to be use if you want to publish in your own helm registry. 
The logic should remain the same and we should still let people upload to their own registry (github repo)

The upload of the chart should be in a different script and in my opinion within CI, we have to stop to manual push packages/images/binaries

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: Let's use fabric like we do in the current release process, to make it cleaner and fits our current tool suite

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: From what I read If I pass `all` as modules it doesn't do what expected but just try to get all-test

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: I don't think specifying the image name is needed. We need to promote all images that have the `imgVersion` available. 

like 
`promote.py feg 1a4b2b30`

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: I think this function should be redesign. 
Calling callJfrogApi with a get shouldn't hardcode get packages. `callJfrogApi` should only be a function that call the api with 0 logic in it just the request.

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: promote  with only the module argument should be able to promote all images of a module.

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: Same here callJfrogApi should only be doing the call, 0 logic there. 
payload could have been defined before for readability

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: As we will implement other promotions here the jfrog url shouldn't be tied to docker

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: This function needs to be reworked to be more simple with 0 promotion logic. 
It should only be a function that calls the Jfrog Api

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: you don't need to print that much information, just the first print is enough

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: combine the 2 prints

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: combine the 2 prints

File: lte/gateway/release/promote.py
Comment (tmdzk) [inline lte/gateway/release/promote.py]: There is some logic missing. 
what if I want to promote all images? 
what if I only want the agw one? 

Let's do fab function:
1 for docker 
1 for agw 
1 for helm

**Review by tmdzk — CHANGES_REQUESTED** [review]
I think most of the code is there. 
But there are a few issues. 

Main feedback related to the promotion script:
1 -  let's use fabric in order to be able to split the differents function and keep using what magma devs and contributors are used to. 
2 - we should be able to promote all the images of a module 
3 - we should be able to promote all images of all modules at once

Please split the upload helm logic into a new script that we will integrate to CI

**Review by hcgatewood — APPROVED** [review]
Approving to unblock need for orc8r-approvers stamp. @tmdzk can handle full review of this PR

As an aside: agree with let's use fab

**Comment by hcgatewood** [comment]
Closing due to inactivity. Please reopen if desired @ranj8work

**Pull request closed without merging**
